tools/target-server-validator/utilities.py

```diff
@@ -268,3 +276,190 @@ def run_parallel(func, args, workers=10):
                 logger.info("No exception information available.")
             logger.error(f"{future} generated an exception")
     return data
+
+
+def get_metric_descriptor(project_id, metric_name):
+    descriptor_name = f"projects/{project_id}/metricDescriptors/{metric_name}"
+
+    client = monitoring_v3.MetricServiceClient()
+    try:
+        descriptor = client.get_metric_descriptor(name=descriptor_name)
+        return descriptor
+    except Exception as e:
+        logger.error(f"Error while getting the existing metric descriptor. ERROR-INFO: {e}")  # noqa
+        return None
+
+
+def create_custom_metric(project_id, metric_name):
+    client = monitoring_v3.MetricServiceClient()
+
+    # Create metric descriptor
+    descriptor = ga_metric.MetricDescriptor()
+    descriptor.type = metric_name
+    descriptor.metric_kind = ga_metric.MetricDescriptor.MetricKind.GAUGE
+    descriptor.value_type = ga_metric.MetricDescriptor.ValueType.DOUBLE
+    descriptor.labels.extend([
+        ga_label.LabelDescriptor(key='hostname', value_type='STRING'),
+        ga_label.LabelDescriptor(key='status', value_type='STRING')
+    ])
+    try:
+        descriptor = client.create_metric_descriptor(name=f"projects/{project_id}", metric_descriptor=descriptor)  # noqa
+        return descriptor
+    except Exception as e:
+        logger.error(f"Error while creating the metric descriptor. ERROR-INFO: {e}")  # noqa
+    return None
+
+
+def get_status_int(status):
```

Comment:
I don't fully understand this. But a float isn't an int :). Would it make sense to create separate status for each?